example/apache-ignite/src/main/scala/org/finos/vuu/example/ignite/schema/IgniteEntitySchema.scala

```diff
@@ -0,0 +1,95 @@
+package org.finos.vuu.example.ignite.schema
+
+import org.apache.ignite.cache.{QueryEntity, QueryIndex}
+import org.finos.vuu.example.ignite.schema.EntitySchema.ColumnName
+import org.finos.vuu.example.ignite.schema.IgniteDataType.{IgniteDataType, fromString}
+import org.finos.vuu.example.ignite.schema.IgniteEntitySchemaBuilder.InvalidIndexException
+import org.finos.vuu.feature.ignite.schema.{ExternalStoreEntitySchema, SchemaField}
+
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+import scala.jdk.CollectionConverters._
+import scala.reflect.runtime.universe.{MethodSymbol, TypeTag, typeOf}
+
+case class IgniteEntitySchema private (private val columnDef: mutable.LinkedHashMap[ColumnName, IgniteDataType],
+                              queryIndex: List[QueryIndex]) extends ExternalStoreEntitySchema {
+
+  private val _schemaFields: List[SchemaField] = columnDef.zipWithIndex.map(
+    {case ((name, dType), i) => SchemaField(name, dType, i)}
+  ).toList
+
+  override def schemaFields: List[SchemaField] = _schemaFields
```

Comment:
Does it make sense to make this a val given that we don't expect use case of ExternalStoreEntitySchema be re-evaluated at the time of gettingSchemaFields? 
If we are trying to solve for the fact you are generating the schemaFields at run time, i think we can solve it using companion object or in this case, do it in the IgniteEntitySchemaBuilder
https://stackoverflow.com/questions/24199490/scala-case-class-generated-field-value

Reply:
Yeah I think it does. I've updated it to instead of overriding a def it overrides a val from the extended trait `ExternalStoreEntitySchema`.

That way we don't need to change the `IgniteEntitySchemaBuilder` and pass in two objects to the `IgniteEntitySchema` where one is derived from the other `fieldDefs` vs `schemaFields`.